core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java

```diff
@@ -291,8 +312,46 @@ private LoadBalancerProvider getProviderOrThrow(String policy, String choiceReas
     return provider;
   }
 
+  /**
+   * Unlike a normal {@link LoadBalancer.Factory}, this accepts a full service config rather than
+   * the LoadBalancingConfig.
+   */
+  ConfigOrError<PolicySelection> selectLoadBalancerPolicy(Map<String, ?> serviceConfig) {
+    try {
+      List<LbConfig> loadBalancerConfigs = null;
+      if (serviceConfig != null) {
+        List<Map<String, ?>> rawLbConfigs =
+            ServiceConfigUtil.getLoadBalancingConfigsFromServiceConfig(serviceConfig);
+        loadBalancerConfigs = ServiceConfigUtil.unwrapLoadBalancingConfigList(rawLbConfigs);
+      }
+      if (loadBalancerConfigs != null && !loadBalancerConfigs.isEmpty()) {
+        List<String> policiesTried = new ArrayList<>();
+        for (LbConfig lbConfig : loadBalancerConfigs) {
+          String policy = lbConfig.getPolicyName();
+          LoadBalancerProvider provider = registry.getProvider(policy);
+          if (provider == null) {
+            policiesTried.add(policy);
+          } else {
+            return ConfigOrError.fromConfig(new PolicySelection(
+                provider,
+                /* serverList= */ null,
+                lbConfig.getRawConfigValue()));
+          }
+        }
+        return ConfigOrError.fromError(
+            Status.UNKNOWN.withDescription(
+                "None of " + policiesTried + " specified by Service Config are available."));
+      }
+      return ConfigOrError.fromError(
+          Status.UNKNOWN.withDescription("Empty Load Balancing config"));
+    } catch (RuntimeException e) {
+      return ConfigOrError.fromError(
+          Status.UNKNOWN.withDescription("can't parse load balancer configuration").withCause(e));
```

Comment:
I think "Empty load balancing config"/"None of policiesTried are available" should be treated differently from "can't parse load balancer configuration". The former should not be considered as service config resolution error.

Reply:
This is existing behavior.  See `decideLoadBalancerProvider`

Reply:
No, it's not the existing behavior. Look at the last `return` of `decideLoadBalancerProvider()`. `defaultPolicy` should be returned if config is empty and there is no balancer address.

Reply:
You are right, I am mistaken.   I changed this to return null because CoE can't wrap null, and a sentinel CoE would be difficult to construct.  

PTAL

Reply:
I think you forgot to push your change.

Reply:
The caller of this method (NrHelper.parseServiceConfig()) doesn't expect `null` from this method. Because `NrHelper.parseServiceConfig()` cannot return `null`, it would have to either return a sentinel result for "no policy configured / use default policy", or just return the default policy directly.

Returning the default policy seems easier, and I would just do it in `selectLoadBalancerPolicy()` instead of `NrHelper.parseServiceConfig()`. It should be outside of the `try-catch`, because the absence of the default policy should be considered a bug rather than a configuration error.

Reply:
There is no default policy, because the default LB is not available yet.  Only the ACLBF knows that.

Reply:
resolved offline

Reply:
The `autoConfiguredLoadBalancerFactory != null` path of `NrHelper.parseServiceConfig()` is not covered in tests. Can you add tests for that?

Reply:
Added a test